src/matter/common/MessageExchange.ts

```diff
@@ -15,9 +15,19 @@ import { Logger } from "../../log/Logger";
 import { NodeId } from "./NodeId";
 import { ByteArray } from "@project-chip/matter.js";
 import { SecureChannelProtocol } from "../session/secure/SecureChannelProtocol";
+import { MatterError } from "../../error/MatterError";
 
 const logger = Logger.get("MessageExchange");
 
+export class UnexpectedMessageError extends MatterError {
+    public constructor(
+        public readonly message: string,
```

Comment:
This one doesn't need to be "public readonly" becauae the base class already define it.